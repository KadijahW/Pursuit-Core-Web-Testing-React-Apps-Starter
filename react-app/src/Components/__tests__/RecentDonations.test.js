import React from 'react';
import { render } from '@testing-library/react';
import RecentDonations from '../RecentDonations';

const sampleDonations = [
  {
    name: 'John',
    caption: 'Here take a break from work',
    amount: 30
  },
  {
    name: 'Emily',
    caption: '',
    amount: 110
  },
  {
    name: 'Sam',
    caption: 'Go to miami',
    amount: 30
  }
]

describe('RecentDonations', () => {
  test('Displays a list of donation cards', () => {
    const { container, getAllByText } = render(<RecentDonations donations={sampleDonations} />)
    expect(getAllByText(/donated/).length).toBe(sampleDonations.length)
    expect(container).toMatchSnapshot()
  })
})
