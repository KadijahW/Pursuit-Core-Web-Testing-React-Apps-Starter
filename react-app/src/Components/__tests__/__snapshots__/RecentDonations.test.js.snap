// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentDonations Displays a list of donation cards 1`] = `
<div>
  <ul
    class="col-4 list-unstyled recent-donations"
  >
    <h5
      class="my-4"
    >
      Recent Donations
    </h5>
    <li
      class="media my-2"
    >
      <div
        class="media-body"
      >
        <h5
          class="mt-0 mb-1"
        >
          John
           donated $
          30
        </h5>
        <p>
          Here take a break from work
        </p>
      </div>
    </li>
    <li
      class="media my-2"
    >
      <div
        class="media-body"
      >
        <h5
          class="mt-0 mb-1"
        >
          Emily
           donated $
          110
        </h5>
        <p />
      </div>
    </li>
    <li
      class="media my-2"
    >
      <div
        class="media-body"
      >
        <h5
          class="mt-0 mb-1"
        >
          Sam
           donated $
          30
        </h5>
        <p>
          Go to miami
        </p>
      </div>
    </li>
  </ul>
</div>
`;
